alternator/streams.cc

```diff
@@ -494,40 +488,78 @@ future<executor::request_return_type> executor::describe_stream(client_state& cl
 
         std::optional<shard_id> last;
 
+        // i is now at the youngest generation we include. make a mark of it.
+        auto first = i;
+
+        // if we're a paged query, skip to the generation where we left of.
+        if (shard_start) {
+            i = topologies.find(shard_start->time);
+        }
+
+        // for parent-child stuff we need id:s to be sorted by token
+        static auto cmp = [](const cdc::stream_id& id1, const cdc::stream_id& id2) {
+            return id1.token() < id2.token();
+        };
+
+        // need a prev even if we are skipping stuff
+        if (i != first) {
+            prev = std::prev(i);
+            std::stable_sort(prev->second.streams.begin(), prev->second.streams.end(), cmp);
```

Comment:
I don't understand this parent-child stuff or what you are sorting or why so it's hard for me to review the validity of this change. Were you able to demonstrate a bug in the old code somehow and this change fixes it? Or how did you test this?

Reply:
Since we cannot force generation changes, we cannot really test this. Even if we can, the parent-child relationship will be ambiguous at best, so I don't know if we can verify it really. The code is mainly just trying to provide a better matching of parent-child via token range of ID:s. Old code used _almost_ but not quite sorted ID:s. 
I've added some more code comments on this in next iteration.

Reply:
See line 515